src/recipes/is-boolean.recipe.ts

```diff
@@ -15,10 +16,12 @@ export function IsBooleanRecipe<Options>(
         // set type to number
         type: 'boolean',
     }, initializers).add(
+        ...buildArrayPropertyDecorators(options.isArray),
+
         // convert strings to boolean
         TypePropertyDecorator(() => Boolean),
 
         // validate data as a boolean
-        IsBoolean(),
+        IsBoolean({ each: !!options.isArray }),
```

Comment:
We must pass `each` to every validator if it is used on an array. Note that some validators take one argument and some take two, so the usage varies slightly. This would have gone horribly wrong without the type checker.

Reply:
Also, to the best of my knowledge `class-transformer` does not have the same need, only the validator.